src/Todo.js

```diff
@@ -1,90 +1,154 @@
-import React, { useState } from "react";
+import React, { useState} from "react";
 import Button from '@mui/material/Button';
 import Grid from '@mui/material/Grid';
-import Modal from '@mui/material/Modal';
-import Box from '@mui/material/Box';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import TableContainer from '@mui/material/TableContainer';
 import { Typography } from "@mui/material";
+// import Checkbox from '@mui/material/Checkbox';
+import BorderColorIcon from '@mui/icons-material/BorderColor';
+import DeleteIcon from '@mui/icons-material/Delete';
+import AddIcon from '@mui/icons-material/Add';
+import UpdateIcon from '@mui/icons-material/Update';
 import Edit from "./edit";
+import Paper from '@mui/material/Paper';
 const TaskList=()=>{
     const[task,setTask]=useState([]);
     const[input,setInput]=useState('');
-
+    const[isEditing,setisEditing]=useState(false);
     const [todoList, setTodoList] = useState([]);
-
+    const [editId,seteditId]=useState();
+    const [isChecked,setIsChecked]=useState(false);
+    const [status,setStatus]=useState(false);
+    //Item add handler
     const addTodo=(text)=>{
         setTask(
             [...task,{ 
                 value:text 
             }]
         );
     }
-
+    
     const inputAdd=(e)=>{
         e.preventDefault();
         console.log("e value",e.target);
         addTodo(input);
     }
-
+    //Delete item Handler
     const deleteItem=(e, id)=>{
         console.log("------- todo id ", id);
         let updatedTODO = todoList.filter( (item, index) => index !== id )
```

Comment:
Don't use filter method for deleting, must use  splice() or split() method from Array for the delete todo Item.